addons/mail_bot/static/src/messaging/widget/notification_alert/notification_alert.js

```diff
@@ -1,23 +1,43 @@
 odoo.define('mail_bot.messaging.widget.NotificationAlert', function (require) {
 "use strict";
 
-var Widget = require('web.Widget');
-var widgetRegistry = require('web.widget_registry');
+const components = {
+    NotificationAlert: require('mail_bot.messaging.component.NotificationAlert'),
+};
+
+const { ComponentWrapper, WidgetAdapterMixin } = require('web.OwlCompatibility');
+
+const Widget = require('web.Widget');
+const widgetRegistry = require('web.widget_registry');
+
+class NotificationAlertWrapper extends ComponentWrapper {}
 
 // -----------------------------------------------------------------------------
 // Display Notification alert on user preferences form view
 // -----------------------------------------------------------------------------
-var NotificationAlert = Widget.extend({
-   template: 'mail_bot.messaging.widget.NotificationAlert',
-   /**
-    * @override
-    */
-   init: function () {
-      this._super.apply(this, arguments);
-      const env = this.call('messaging', 'getEnv');
-      const hasRequest = env.store.state.mailbotHasRequest;
-      this.isNotificationBlocked = window.Notification && window.Notification.permission !== "granted" && !hasRequest;
-   },
+const NotificationAlert = Widget.extend(WidgetAdapterMixin, {
+    /**
+     * @override
+     */
+    init() {
+        this._super(...arguments);
+        this.component = undefined;
+    },
+    /**
+     * @override
+     */
+    async start() {
+        this._super(...arguments);
+        this.env = this.call('messaging', 'getEnv');
```

Comment:
Storing `env` on `this` does not seem necessary here, unless that is needed by `WidgetAdapterMixin`?